README.md

```diff
@@ -17,3 +17,329 @@ Implement a prototype job worker service that provides an API to run arbitrary L
 
 * Client should be able to query result of the job execution and fetch the logs
 
+## Design Document
+
+### Limitations And Scope
+
+#### Data Management
+Though the use of a database to store persistant data would be ideal, I will be instead storing the outputs and error outputs into logs stored on 
+the file storage of the linux worker. The logs will be generated at start time with the foldername \<pid\>_\<startingTimestamp\>
+The client will have to maintain a queue of commands, the client can query the server to see what is running to determine what can be killed,
+or the client can store response data from server.
+
+#### Scale
+The scope of this project would only deal with a single linux worker server interfacing with one client
+
+### API Design
+
+* The client will have an Command Api that takes in three types of request, Start, Stop, and Query.
+
+* The client can then execute the commands over the server via Execute Api
+
+```
+type CommandRequest {
+    variableTypeRequest {
+        StartRequest start 
+        StopRequest stop 
+        QueryRequest query 
+    }
+}
+
+type CommandResponse {
+    variableField response {
+        StartResponse responseFromStart 
+        StopResponse responseFromStop 
+        QueryResponse responseFromQuery 
+    }
+}
+
+
+func Execute(CommandRequest) returns(CommandResponse)
+
+```
+
+#### Start
+
+* The Start command is called with a StartRequest that has the client's command and required arguments and optional env, dir params
+
+* The Process be executed, three logs will be generated, the standard output and standard error will be redirected into a log file, one for stdout and another for stderr, the third log called programName.log will have the name of the program and arguments executed in the contents of the log. All logs will be located in the same folder with the foldername \<pid\>_\<startingTimestamp\>, a response with the pid is sent, outputs should be empty since it was just started
```

Comment:
Long term log persistence is not a hard requirement, so if you find it easier to store logs in-memory that is totally OK.

Reply:
will definitely consider this, and think about more